src/GitHub.Api/Application/ApplicationManagerBase.cs

```diff
@@ -170,28 +193,43 @@ private void SetupGit(GitInstaller.GitInstallationState state)
 
         public void InitializeRepository()
         {
+            isBusy = true;
             var thread = new Thread(obj =>
             {
+                var success = true;
                 CancellationToken token = (CancellationToken)obj;
-                var targetPath = NPath.CurrentDirectory;
-
-                var gitignore = targetPath.Combine(".gitignore");
-                var gitAttrs = targetPath.Combine(".gitattributes");
-                var assetsGitignore = targetPath.Combine("Assets", ".gitignore");
-
-                var filesForInitialCommit = new List<string> { gitignore, gitAttrs, assetsGitignore };
+                try
+                {
+                    var targetPath = NPath.CurrentDirectory;
+
+                    var gitignore = targetPath.Combine(".gitignore");
+                    var gitAttrs = targetPath.Combine(".gitattributes");
+                    var assetsGitignore = targetPath.Combine("Assets", ".gitignore");
+
+                    var filesForInitialCommit = new List<string> { gitignore, gitAttrs, assetsGitignore };
+
+                    GitClient.Init().RunWithReturn(true);
+                    GitClient.LfsInstall().RunWithReturn(true);
+                    AssemblyResources.ToFile(ResourceType.Generic, ".gitignore", targetPath, Environment);
+                    AssemblyResources.ToFile(ResourceType.Generic, ".gitattributes", targetPath, Environment);
+                    assetsGitignore.CreateFile();
+                    GitClient.Add(filesForInitialCommit).RunWithReturn(true);
+                    GitClient.Commit("Initial commit", null).RunWithReturn(true);
+                    Environment.InitializeRepository();
+                    UsageTracker.IncrementProjectsInitialized();
```

Comment:
Do you want to do this an an `ActionTask` exeuction, "fire and forget" style?

Reply:
The whole repo initialization takes a while, so doing this in thread is not a big deal I think.